akka-cluster/src/multi-jvm/scala/akka/cluster/StreamRefSpec.scala

```diff
@@ -0,0 +1,247 @@
+      }
+      enterBarrier("actor-started")
+
+      // only used from first
+      var destinationForSource: TestSubscriber.Probe[String] = null
+
+      runOn(first) {
+        system.actorSelection(node(second) / "user" / "dataSource") ! Identify(None)
+        val ref = expectMsgType[ActorIdentity].ref.get
+        ref ! RequestLogs(1337)
+        val dataSourceRef = expectMsgType[LogsOffer].sourceRef
+        destinationForSource = dataSourceRef.runWith(TestSink.probe)
+        destinationForSource
+          .request(3)
+          .expectNext("elem-1")
+          .expectNext("elem-2")
+          .expectNext("elem-3")
+      }
+      runOn(second) {
+        dataSourceLifecycle.expectMsg("started-1337")
+      }
+      enterBarrier("streams-started")
+
+      runOn(first) {
+        testConductor.blackhole(first, second, Direction.Both).await
+        testConductor.blackhole(third, second, Direction.Both).await
+      }
+      enterBarrier("after-split")
+
+      // auto-down
+      runOn(first, third) {
+        awaitMembersUp(2, Set(second) map address)
+      }
+      runOn(second) {
+        awaitMembersUp(1, Set(first, third) map address)
+      }
+      enterBarrier("members-removed")
+
+      runOn(first) {
+        destinationForSource.expectError().getClass should ===(classOf[RemoteStreamRefActorTerminatedException])
+      }
+      runOn(second) {
+        dataSourceLifecycle.expectMsg("ended-1337")
+      }
+
+      enterBarrier("after-2")
+    }
+
+    "stop stream with SinkRef after downing and removal" taggedAs LongRunningTest in {
+      import system.dispatcher
+      val streamLifecycle1 = TestProbe()
+      val streamLifecycle3 = TestProbe()
+      runOn(third) {
+        system.actorOf(DataReceiver.props(streamLifecycle3.ref), "dataReceiver")
+      }
+      enterBarrier("actor-started")
+
+      runOn(first) {
+        system.actorSelection(node(third) / "user" / "dataReceiver") ! Identify(None)
+        val ref = expectMsgType[ActorIdentity].ref.get
+        ref ! PrepareUpload("system-42-tmp")
+        val ready = expectMsgType[MeasurementsSinkReady]
+
+        Source(1 to 1000)
+          .map(n ⇒ s"elem-$n")
+          .watchTermination()(Keep.right)
+          .to(ready.sinkRef)
+          .run()
+          .onComplete(_ ⇒ streamLifecycle1.ref ! "ended-system-42-tmp")
+      }
+      runOn(third) {
+        streamLifecycle3.expectMsg("started-system-42-tmp")
+      }
+      enterBarrier("streams-started")
+
+      runOn(first) {
+        testConductor.blackhole(first, third, Direction.Both).await
+      }
+      enterBarrier("after-split")
+
+      // auto-down
+      runOn(first) {
+        awaitMembersUp(1, Set(third) map address)
+      }
+      runOn(third) {
+        awaitMembersUp(1, Set(first) map address)
+      }
+      enterBarrier("members-removed")
+
+      runOn(first) {
+        streamLifecycle1.expectMsg("ended-system-42-tmp")
```

Comment:
Wouldn't it be better to verify that blackhole+down specifically lead to failing the stream?